app/services/sign_in/assertion_validator.rb

```diff
@@ -32,7 +34,9 @@ def validate_iss
     end
 
     def validate_audience
-      unless decoded_assertion.aud.match(token_route)
+      audience = decoded_assertion.aud&.class == Array ? decoded_assertion.aud.first : decoded_assertion.aud
```

Comment:
What you'll want to do here is first wrap the input in Array:
`Array(decoded_assertion.aud)`, this will convert it to an array regardless of the content. Then you'll be able to perform the check with the assumption that the object is an array

Reply:
And then off the array I believe you should be able to use `.include?` to match the audience